### PR TITLE
Feature/contextual

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
-  Add contextual app bar. 

Some questions:
Q1:  For mobile, this should be a table, right?  Do we want the Title to always be aligned with the 'Name' cell?  I might need to refactor this to be al list instead or just introduce fixed cell sizes. 
change-size

Q2:  Empty state when all items are deleted? 
image

@huayunh

### CODE REVIEW

**Comment by jeffvg** [comment]
We've been trying to include empty states where it makes sense and in this case it does since we can delete all the content. So maybe the empty state includes button to add a new row or simply reload what was there

**Comment by huayunh** [comment]
If you want to go fancy about it,


Since this is not a table pattern (yet), I would suggest a flex 0 for the checkbox, and flex 1 for the other two columns

**Review by huayunh — COMMENTED** [review]
# 1
image
when something rows are checked (but not every row), the top left corner should be in "indeterminate" state. Clicking on an indeterminate checkbox shall either select all or unselect all, your call.

# 2
Would you also make the delete button red, and write the button text Title Case? I've updated the design file.

# 3
On desktop (and tablet size), Would you also add a text at the bottom of the table that says "The contextual app bar is for mobile only."? I've updated the design. Sorry for that.

# 4 
Can we apply a Blue500 5% background color to selected rows?

# 5
To your Q2, you can do something like this
image
That will re-populate the table content.

**Comment by emclaug2** [comment]
@huayunh  This should be ready for another round of review; the page has been updated with the new specs & added the missing ones.

File: src/app/pages/app-bar/contextual/contextual.component.scss
Comment (huayunh) [reply src/app/pages/app-bar/contextual/contextual.component.scss:66]: I also suggest give the checkbox column a fixed width so that: 
image

We will probably revisit some of these styles in the future once we have the tables pattern rolled out on docit

File: src/app/pages/app-bar/contextual/contextual.component.scss
Comment (huayunh) [inline src/app/pages/app-bar/contextual/contextual.component.scss:56]: It appears that the background opacity is overlapping. May want to do something like 

```suggestion
            background-color: rgba(map-get($pxb-blue, 500), 0.05), map-get($pxb-white, 50);
```

image

**Review by huayunh — APPROVED** [review]

**Comment by jeffvg** [comment]
I noticed comment above show gif of the scrolling but did we intend to have horizontal scroll when we snap down to 366 width and on small device view?

image

image

**Comment by huayunh** [comment]
And as a reminder, don't forget to update the Miro board after things get merged to track our progress

File: src/app/pages/app-bar/contextual/contextual.component.html
Comment (emclaug2) [reply src/app/pages/app-bar/contextual/contextual.component.html:1]: 2 upped to 4

File: src/app/pages/app-bar/contextual/contextual.component.scss
Comment (emclaug2) [reply src/app/pages/app-bar/contextual/contextual.component.scss:63]: You got it.

File: src/app/pages/app-bar/contextual/contextual.component.scss
Comment (emclaug2) [reply src/app/pages/app-bar/contextual/contextual.component.scss:66]: > 16px

This was 20px to align the checkboxes with the close & menu icon.   I've added a fixed width the cell to align Name & Toolbar title.

**Comment by emclaug2** [comment]
> I noticed comment above show gif of the scrolling but did we intend to have horizontal scroll when we snap down to 366 width and on small device view?
> 
> image
> 
> image

I think the table should overflow but the text below the table should not be a part of that overflowing element.  I'll move the message so it wraps instead of scrolls with the table.

File: src/app/pages/app-bar/contextual/contextual.component.scss
Comment (emclaug2) [reply src/app/pages/app-bar/contextual/contextual.component.scss]: The sticky cell was overlapping each table's row, so we got a double background.  This should be fixed now.

**Comment by emclaug2** [comment]
Branch should be good now.

**Pull request merged**
